src/Symfony/Maker/Resources/skeleton/StateProcessor.tpl.php

```diff
@@ -8,8 +8,8 @@
 
 class <?php echo $class_name; ?> implements ProcessorInterface
 {
-    public function process(mixed $data, Operation $operation, array $uriVariables = [], array $context = []): void
+    public function process(mixed $data, Operation $operation, array $uriVariables = [], array $context = []): mixed
     {
-        // Handle the state
+        // Handle the state and return the inner processor
```

Comment:
this is only in case of decorating another provider, a provider can definitely return void.